### PR TITLE
Add TokenExpiration to Response

### PR DESCRIPTION
**Pull request by w-haibara**
GitHub has been updated to allow you to set the expiration date of your personal access token. And the expiration date is now displayed in the `GitHub-Authentication-Token-Expiration` header of the GItHub API. 
https://github.blog/changelog/2021-07-26-expiration-options-for-personal-access-tokens/

This PR add `TokenExpiration` field to Response.

### CODE REVIEW

File: example/tokenauth/main.go
Comment (gmlewis) [inline example/tokenauth/main.go:47]:
```suggestion
		log.Printf("Token Expiration: %v\n", resp.TokenExpiration)
```

File: github/github.go
Comment (gmlewis) [inline github/github.go:564]:
```suggestion
		if t, err := time.Parse("2006-01-02 03:04:05 MST", v); err == nil {
			exp = Timestamp{t.Local()}
		}
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @w-haibara !

In addition to the comments below, please add a new unit test for the new `parseTokenExpiration` function.

Also please make sure to run `gofmt` on the edited files so that their formatting is standardized.

See CONTRIBUTING.md in the top-level of this repo for more information.

**Comment by w-haibara** [comment]
Hi, @gmlewis ! Thanks for the thoughtful review.
I've fixed the areas you commented on and added a test. I apologize again for the unformatted commits. I've fixed that as well.

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:2229]: Can you please add one more test with a completely garbage header and show how it behaves in that case?

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:2225]: Since `res` starts out the same in every test case, you can remove it from this struct and initialize a new fresh one below within the test loop.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @w-haibara !

**Review by gmlewis — APPROVED** [review]
Thank you, @w-haibara !
LGTM.

Awaiting second LGTM before merging.

**Comment by w-haibara** [comment]
Thank you, @gmlewis.
This is almost the first OSS contributor for me. Your review has helped me a lot!

**Comment by gmlewis** [comment]
> Thank you, @gmlewis.
> This is almost the first OSS contributor for me. Your review has helped me a lot!

Thank you, for your contributions, @w-haibara !  Your help is greatly appreciated!

**Review by Parker77 — APPROVED** [review]
LGTM!

**Pull request merged**
